packages/babel-plugin-material-ui/src/index.ts

```diff
@@ -0,0 +1,117 @@
+import * as babel from '@babel/core';
+import * as t from '@babel/types';
+import resolveImports, { MappedImportsType, ImportType } from './resolveImports';
+import { Packages } from './packages';
+
+/**
+ * Holds the import name and location of everything found in the index file of each package
+ */
+const importMaps: { [K: string]: MappedImportsType | undefined } = {};
+
+/**
+ * Used to transform the specified root import to specific imports based on the exports found
+ * in the index file of the package
+ */
+function transformImport(path: babel.NodePath<t.ImportDeclaration>, packageName: string) {
+  if (importMaps[packageName] === undefined) {
+    importMaps[packageName] = resolveImports(packageName);
+  }
+
+  const { node } = path;
+  const importMap = importMaps[packageName]!;
+
+  for (const key in importMap) {
```

Comment:
Prefer explicitly iterating over properties with e.g. Object.keys. This would be rejected by the linter anyway (which doesn't run on TS currently).